lms/djangoapps/grades/new/subsection_grade.py

```diff
@@ -104,51 +93,66 @@ def load_from_data(self, model, course_structure, scores_client, submissions_sco
-            self.all_total.earned,
-            self.all_total.possible,
-            self.graded_total.earned,
-            self.graded_total.possible,
-            self.location,
-            self.display_name
+        return PersistentSubsectionGrade.bulk_create_grades(
+            [subsection_grade._persisted_model_params(student) for subsection_grade in subsection_grades],  # pylint: disable=protected-access
+            course_key,
         )
 
+    def create_model(self, student):
+        """
+        Saves the subsection grade in a persisted model.
+        """
+        self._log_event(log.info, u"create_model", student)
+        return PersistentSubsectionGrade.create_grade(**self._persisted_model_params(student))
+
+    def update_or_create_model(self, student):
+        """
+        Saves or updates the subsection grade in a persisted model.
+        """
+        self._log_event(log.info, u"update_or_create_model", student)
+        return PersistentSubsectionGrade.update_or_create_grade(**self._persisted_model_params(student))
+
     def _compute_block_score(
             self,
             student,
             block_key,
             course_structure,
             scores_client,
             submissions_scores,
-            persisted_values=None,
+            persisted_values,
     ):
         """
-        Compute score for the given block. If persisted_values is provided, it will be used for possible and weight.
+        Compute score for the given block. If persisted_values
+        is provided, it is used for possible and weight.
         """
         block = course_structure[block_key]
 
         if getattr(block, 'has_score', False):
+
+            possible = persisted_values.get('possible', None)
+            weight = persisted_values.get('weight', getattr(block, 'weight', None))
+
             (earned, possible) = get_score(
                 student,
                 block,
                 scores_client,
                 submissions_scores,
+                weight,
+                possible,
             )
 
-            # There's a chance that the value of weight is not the same value used when the problem was scored,
-            # since we can get the value from either block_structure or CSM/submissions.
-            weight = getattr(block, 'weight', None)
-            if persisted_values:
-                possible = persisted_values.get('possible', possible)
-                weight = persisted_values.get('weight', weight)
-
             if earned is not None or possible is not None:
+                # There's a chance that the value of graded is not the same
+                # value when the problem was scored. Since we get the value
+                # from the block_structure.
+                #
                 # cannot grade a problem with a denominator of 0
                 block_graded = block.graded if possible > 0 else False
```

Comment:
if earned is not None, possible can still be None, and then we're comparing `None` to `0`, which is problematic.  (And will fail outright in python3).